apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PooledSession.cs

```diff
@@ -231,6 +231,34 @@ public void ReleaseToPool(bool forceDelete)
             }
         }
 
+        /// <summary>
+        /// Detaches this session from the session pool from which it was acquired, so that the session pool
+        /// stops tracking this session and counting it as active.
+        /// This method should only be called once per instance; subsequent calls are ignored.
+        /// No other methods can be called after this.
+        /// </summary>
+        /// <remarks>
+        /// This method should be called only for sessions that are meant to be explicitly shared across processes.
+        /// Note that we don't attempt to rollback a transaction that is being detached, or attempt to delete the session,
+        /// under the assumption that it will be reused across processes.
+        /// If there's a process capable of knowing when all other processes are done using the session, then that process could call
+        /// <see cref="SessionPool.CreateDetachedSession(SessionName, ByteString, ModeOneofCase)"/> (or an overload) to create an instance
+        /// of <see cref="PooledSession"/> representing the shared transaction and then call <see cref="ReleaseToPool(bool)"/> passing true
+        /// to force session deletion and clean up resources.
+        /// Else, the application can rely on Spaner service garbage collection to clean up this session once it becomes stale.
+        /// </remarks>
+        public void DetachFromPool()
+        {
+            if (MarkAsDisposed())
+            {
+                _pool.Detach(AfterReset());
```

Comment:
Why are we calling `AfterReset()` here? Won't that remove the transaction from the session, which effectively makes it unusable from this point on? Or am I missing something here?

Reply:
(Sorry for the slow response, I'm on-call this week).

Sorry, I thought this method was called already when you would call `Transaction.DisposeBehavior = Detach`. That is not correct, it is called when the transaction is disposed. And then it is obvious that you can't use the transaction anymore. Thanks for the explanation.